### ISSUE TITLE
Add accessibility tests via jest

### ISSUE THREAD

**Issue by albullington** [inaturalist/iNaturalistReactNative#308]
Eventually, we're going to want accessibility tests written for all unit tests & integration tests. They should be pretty straightforward, like the test in `ObsCard.test.js`:

```
test( "should not have accessibility errors", ( ) => {
  const obsCard = <ObsCard item={testObservation} />;

  expect( obsCard ).toBeAccessible( );
} );
```

- [x] All unit tests/integration tests should have a test for accessibility
- [x] The test should render a component and check for accessibility using the matcher from `react-native-accessibility-engine`
- [x] All relevant errors should be fixed (adding accessibility labels, etc.)

We might run into some issues here with third party libraries not letting us add accessibility labels. If this happens, it's probably fine to mark a test as todo for now.
